packages/openwrt/Makefile

```diff
@@ -51,50 +30,88 @@ define Package/n2n-edge
   DEPENDS+=+kmod-tun
 endef
 
+define Package/n2n-edge/description
+  The client node for the N2N infrastructure
+endef
+
+define Package/n2n-edge/conffiles
+/etc/n2n/edge.conf
+endef
+
 define Package/n2n-supernode
   $(call Package/n2n/Default)
   TITLE+= server (supernode)
 endef
 
-define Package/n2n-edge/description
-The client node for the N2N infrastructure
+define Package/n2n-supernode/description
+  The supernode for the N2N infrastructure
 endef
 
-define Package/n2n-supernode/description
-The supernode for the N2N infrastructure
+define Package/n2n-supernode/conffiles
+/etc/n2n/supernode.conf
 endef
 
-define Build/Configure
-	( cd $(PKG_BUILD_DIR); \
-	./autogen.sh; \
-	./configure )
+define Package/n2n-tests
+  $(call Package/n2n/Default)
+  TITLE+= tests
 endef
 
-define Package/n2n-edge/conffiles
-/etc/n2n/edge.conf
+define Package/n2n-tests/description
+  The tests for the N2N infrastructure
 endef
 
+define Package/n2n-utils
+  $(call Package/n2n/Default)
+  TITLE+= utilities
+  DEPENDS+=+libpcap
+endef
+
+define Package/n2n-utils/description
+  The extend utilities for the N2N infrastructure
+endef
+
+CMAKE_OPTIONS+= \
+	-DCMAKE_BUILD_TYPE=Release \
+	-DN2N_OPTION_USE_CAPLIB=ON \
+	-DN2N_OPTION_USE_PCAPLIB=ON \
+	-DN2N_OPTION_USE_ZSTD=ON
+
 define Package/n2n-edge/install
 	$(INSTALL_DIR) $(1)/usr/bin
-	$(INSTALL_BIN) $(PKG_BUILD_DIR)/edge $(1)/usr/bin/
+	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/sbin/edge $(1)/usr/bin/n2n-edge
 	$(INSTALL_DIR) $(1)/etc/init.d
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/packages/openwrt/etc/init.d/edge $(1)/etc/init.d/edge
 	$(INSTALL_DIR) $(1)/etc/n2n
 	$(INSTALL_CONF) $(PKG_BUILD_DIR)/packages/etc/n2n/edge.conf.sample $(1)/etc/n2n/edge.conf
 endef
 
-define Package/n2n-supernode/conffiles
-/etc/n2n/supernode.conf
-endef
-
 define Package/n2n-supernode/install
 	$(INSTALL_DIR) $(1)/usr/bin
-	$(INSTALL_BIN) $(PKG_BUILD_DIR)/supernode $(1)/usr/bin/
+	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/sbin/supernode $(1)/usr/bin/n2n-supernode
 	$(INSTALL_DIR) $(1)/etc/init.d
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/packages/openwrt/etc/init.d/supernode $(1)/etc/init.d/supernode
 	$(INSTALL_DIR) $(1)/etc/n2n
 	$(INSTALL_CONF) $(PKG_BUILD_DIR)/packages/etc/n2n/supernode.conf.sample $(1)/etc/n2n/supernode.conf
 endef
 
+define Package/n2n-tests/install
+	$(INSTALL_DIR) $(1)/usr/bin
+	$(CP) $(PKG_BUILD_DIR)/tests/tests_units.list $(1)/usr/bin/n2n-tests_units.list
+	( for i in $$$$(grep -v '#' $(PKG_BUILD_DIR)/tests/tests_units.list); do \
+		$(INSTALL_BIN) $(PKG_BUILD_DIR)/$$$$i $(1)/usr/bin/n2n-$$$$i ; \
+	done )
+endef
+
+define Package/n2n-utils/install
+	$(INSTALL_DIR) $(1)/usr/bin
+	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/n2n-benchmark $(1)/usr/bin/
+	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/n2n-decode $(1)/usr/bin/
+	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/n2n-keygen $(1)/usr/bin/
+	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/n2n-portfwd $(1)/usr/bin/
+	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/n2n-route $(1)/usr/bin/
```

Comment:
If you are splitting this into small packages for embedded use, It seems that some of these utils can be used by the daemon in regular operations and some are only used for testing or configuration.  So, this may not be the best set to install in each package

Reply:
I'm not familiar with them. It would be best if you can give me some advice.